src/main/java/seedu/address/model/patient/Name.java

```diff
@@ -35,7 +36,7 @@ public Name(String name) {
      * Returns true if a given string is a valid name.
      */
     public static boolean isValidName(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return test.matches(VALIDATION_REGEX) && test.length() <= 150;
```

Comment:
An alternative is to check using regex. `[\\p{Alnum}]{1, 150}` should do the trick.